src/messages/responses/set-fetch.ts

```diff
@@ -0,0 +1,37 @@
+import {ResponseBase} from './response-base';
+import {SdkError} from '../../errors/errors';
+import {applyMixins, ErrorBody} from '../../errors/error-utils';
+import {TextDecoder} from 'util';
+
+const TEXT_DECODER = new TextDecoder();
+
+export abstract class Response extends ResponseBase {}
+
+export class Missing extends Response {}
+
+export class Found extends Response {
+  private readonly elements: Uint8Array[];
+
+  constructor(elements: Uint8Array[]) {
+    super();
+    this.elements = elements;
+  }
+
+  public valueSetString(): Set<string> {
+    return new Set(this.elements.map(e => TEXT_DECODER.decode(e)));
```

Comment:
Hopefully `TextDecoder.decode` is copy-on-write.

Reply:
@schwern what does it mean? and why do we want copy-on-write here? (I'm not familiar with the concept...)

Reply:
If someone tries to tamper with the underlying byte array (which I think they can get via `valueSetByteArray()`) ,then will that tampering change the values returned from `valueSetString()`?

I think that's what he's trying to get at.

Reply:
Gonna write some unit tests to verify this.

Reply:
Ohhh, I see. Thank you for the explanation, Vir!

Reply:
Copy-on-write (COW) is where multiple variables share the same underlying memory until one of them changes.

The primary purpose is to save memory. In this case, TextDecoder.decode shouldn't need to copy the values just to change their encoding. `this.elements` and the returned `Set<string>` should share the same memory.

@virratanasangpunth It's an optimization, it shouldn't be detectable in tests.

```
str1 = "test"

# without COW the string is copied here
# with COW, the string is not copied. Both str1 and str2 share the same underlying memory for the string.
str2 = str1

# with COW, now str2 makes a copy and changes it.
str2.concat(" more")
```

Reply:
Hmmm I'm still exploring this topic in my local branch.

I think I"m gonna merge the PR as-is and I can come back to optimize/fix (if needed) once I'm done exploring.

Reply:
I wouldn't worry about it. It's hidden from the user. We can always optimize later.

Reply:
> It's hidden from the user

I think that's the part where I'm not 100% convinced :) but you're probably right.